README.md

```diff
@@ -1,189 +1,785 @@
+   (`BlockstackSignInButton`). You use this button in your here with the
+   `org.blockstack.android.sdk.ui.BlockstackSignInButton` class.
 
-session.getFile(fileName, options, {content: Any ->
-  // content can be a `String` or a `ByteArray`
+    ```XML
+    <?xml version="1.0" encoding="utf-8"?>
+    <android.support.constraint.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:app="http://schemas.android.com/apk/res-auto"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
+    tools:context=".MainActivity">
+
+    <org.blockstack.android.sdk.ui.BlockstackSignInButton
+        android:id="@+id/signInButton"
+        android:layout_width="307dp"
+        android:layout_height="45dp"
+        android:layout_margin="4dp"
+        android:layout_marginEnd="185dp"
+        android:layout_marginStart="39dp"
+        app:layout_constraintEnd_toEndOf="parent"
+        app:layout_constraintStart_toStartOf="parent"
+        tools:layout_editor_absoluteY="16dp" />
+
+    <TextView
+        android:id="@+id/userDataTextView"
+        android:layout_width="370dp"
+        android:layout_height="27dp"
+        tools:layout_editor_absoluteX="6dp"
+        tools:layout_editor_absoluteY="70dp" />
+
+    </android
+    ```
+
+    This codes adds a button and some text to your application.
+
+4. Choose **Run > Apply changes**.
+
+5. Choose **Run > Run app** in the emulator.
+
+    The emulator now contains a new interface with a button:
+
+    ![](images/new-interface.png)
+
+### Add session & authentication code
+
+1. Open the `MainActivity.kt` file.
+2. Add some additional imports to the top below the `android.os.Bundle` import.
+
+    When you are done, your imports should appear as follows:
+
+    ```kotlin
+
+    import android.support.v7.app.AppCompatActivity
+    import android.os.Bundle
+
+    import android.support.v7.app.AppCompatActivity
+    import android.view.View
+    import kotlinx.android.synthetic.main.activity_main.*
+    import org.blockstack.android.sdk.BlockstackSession
+    import org.blockstack.android.sdk.Scope
+    import org.blockstack.android.sdk.UserData
+    import java.net.URI
+    ```
+3. Add a variable for the Blockstack session before `onCreate`.
+
+   ```kotlin
+   class MainActivity : AppCompatActivity() {
+
+    private var _blockstackSession: BlockstackSession? = null
+
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+      }
+    }
+   ```
+
+4. Replae the existing the `onCreate` function with the following:
+
+    ```kotlin
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_main)
+
+        signInButton.isEnabled = false
+
+        val appDomain = URI("https://flamboyant-darwin-d11c17.netlify.com")
```

Comment:
Add comment to change the appDomain

Reply:
See the previous comment.  I'm not adding in code comments for the connection info, it was deliberately left as this.  We want them to have a working connection for the first pass.